Dockerfile

```diff
@@ -1,11 +1,17 @@
-FROM python:3.12.3-slim-bullseye
+FROM python:3.12.3-slim-bookworm
 
 RUN adduser --disabled-password terraso
 
 ENV PATH /home/terraso/.local/bin:$PATH
 # see https://github.com/aws/aws-cli/tags for list of versions
 ENV AWS_CLI_VERSION 2.8.12
 
+# Add testing sources and pin the GDAL packages to testing
+# Allows us to get 3.8.x versions of GDAL
+RUN sed 's/bookworm/testing/g' /etc/apt/sources.list.d/debian.sources >  /etc/apt/sources.list.d/testing.sources
+
+RUN echo 'Package: libgdal-dev gdal-bin\nPin: release a=testing\nPin-Priority: 900' > /etc/apt/preferences.d/gdal-testing
```

Comment:
question: is there a more structured `apt` command to do this rather than manually inserting text into `apt` config files?

Reply:
I looked, and couldn't find one.

Reply:
ok, i also don't understand why we aren't using gdal 3.6.x to avoid these incompatibilities like we discussed last week but this seems fine for now

Reply:
Everyone's dev environment has 3.8.x, so I wanted to match that.